iothub/service/src/IotHubTokenCredential.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using System.Threading;
+
+#if !NET451
+
+using Azure.Core;
+
+#endif
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// The properties required for authentication to IoT hub using a token credential.
+    /// </summary>
+    internal class IotHubTokenCredential : IotHubCredential
+    {
+#if !NET451
+
+        private const string _tokenType = "jwt";
+        private TokenCredential _credential;
+
+        public IotHubTokenCredential(string hostName, TokenCredential credential) : base(hostName)
+        {
+            _credential = credential;
+        }
+
+#endif
+
+        public override string GetAuthorizationHeader()
+        {
+#if !NET451
+            AccessToken token = _credential.GetToken(new TokenRequestContext(), new CancellationToken());
+            return $"Bearer {token.Token}";
+#else
+            throw new InvalidOperationException($"{nameof(GetAuthorizationHeader)} is not supported on NET451");
```

Comment:
Q - is it worth adding the class name into the exception message => signifying that TokenCredential is the type that isn't supported in net451?

Reply:
I believe it should not even get here. I am now throwing an exception in the constructor itself that TokenCredential is not supported on NET451.